controllers/usersignup/usersignup_controller.go

```diff
@@ -840,6 +856,18 @@ func (r *Reconciler) ensureSpace(logger logr.Logger, userSignup *toolchainv1alph
 	return space, true, nil
 }
 
+func (r *Reconciler) ensureSpaceReady(logger logr.Logger, space *toolchainv1alpha1.Space) (bool, error) {
+	logger.Info("Ensuring Space is Ready", "Space", space.Name)
+	readyCond, ok := condition.FindConditionByType(space.Status.Conditions, toolchainv1alpha1.ConditionReady)
+	if ok {
+		if readyCond.Status == corev1.ConditionTrue && readyCond.Reason == toolchainv1alpha1.SpaceProvisionedReason {
+			return true, nil
+		}
+		return false, nil
+	}
+	return false, fmt.Errorf("ready condition not found on space %s", space.Name)
+}
```

Comment:
Since we're only checking the status of the space and not actually changing anything I think renaming this func to something like `isSpaceReady` is more appropriate and it can probably be simplified to just return a `bool` because even if the Space isn't ready it's not necessarily a problem since we expect it to take some time. Otherwise we'd be printing errors in the logs under normal operating conditions.

```suggestion
func (r *Reconciler) isSpaceReady(logger logr.Logger, space *toolchainv1alpha1.Space) (bool, error) {
	logger.Info("Checking whether Space is Ready", "Space", space.Name)
	readyCond, ok := condition.FindConditionByType(space.Status.Conditions, toolchainv1alpha1.ConditionReady)
	return ok && readyCond.Status == corev1.ConditionTrue && readyCond.Reason == toolchainv1alpha1.SpaceProvisionedReason
}
```

Reply:
I like the idea of just returning the bool. Lemme make that change! Thanks!

Reply:
done in [bc13ccb](https://github.com/codeready-toolchain/host-operator/pull/670/commits/bc13ccbc3af4a5f2e63d48f1cd695476e0e516aa)